### PR TITLE
PREVIEW: Wallofcoins buying wizard

### PR DESCRIPTION
**Pull request by rgenito**
Heads up!

We will have issues that we are resolving in QA. This is a PREVIEW of the branch merge.

### CODE REVIEW

**Comment by HashEngineering** [comment]
There are many merge conflicts with this request because there have been many commits to the master branch.  A rebase will be required before any pull requests are tested or approved.

**Comment by HashEngineering** [comment]
When we do merge this we will flatten the request into a single commit.

File: README.md [hunk 1]
```diff
@@ -1,40 +1,329 @@
-Welcome to _Dash Wallet_, a standalone Dash payment app for your Android device!
+### WOC Buy API Setup (Android)
 
-This project contains several sub-projects:
```
Comment (HashEngineering) [inline README.md]: This file should not be changed.  This is the main read me for the repo.  Wall Of Coins api specific information should go in a separate file.

**Comment by rgenito** [comment]
Sorry about that! I’ll let Viral know.

File: build.gradle [hunk 2]
```diff
@@ -1,15 +1,17 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.2'
```
Comment (HashEngineering) [inline build.gradle]: In the current version, we are using version 2.10, the same as Bitcoin Wallet, because that allows the NDK features to work for scrypt and X11 native hashing.

File: settings.gradle [hunk 3]
```diff
@@ -1,3 +1,3 @@
 include 'wallet'
 include 'integration-android'
-include 'sample-integration-android'
```
Comment (HashEngineering) [inline settings.gradle]: This looks like it was done to remove annoyances from this example, but it has nothing to do with wall of coins support.

File: wallet/AndroidManifest.xml [hunk 4]
```diff
@@ -1,244 +1,257 @@
...
-			android:exported="false" />
-
-		<service
-			android:name="de.schildbach.wallet.service.InactivityNotificationService"
-			android:exported="false" />
-
-		<meta-data
-			android:name="android.nfc.disable_beam_default"
-			android:value="true" />
-	</application>
```
Comment (HashEngineering) [inline wallet/AndroidManifest.xml]: Don’t make white space changes to this file.  We need to keep it the same as Bitcoin Wallet.

File: wallet/res/values/strings.xml [hunk 5]
```diff
@@ -1,356 +1,396 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <resources>
```
Comment (HashEngineering) [inline wallet/res/values/strings.xml]: First, we don’t want whitespace changes.

Secondly, no Wall of Coins strings should be in this file.  They should be in strings-extra.xml to avoid conflicts with Bitcoin Wallet merges.

File: wallet/src/de/schildbach/wallet/ExchangeRatesProvider.java [hunk 6]
```diff
@@ -52,10 +52,14 @@
 import android.preference.PreferenceManager;
 import android.provider.BaseColumns;
 import android.text.format.DateUtils;
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ExchangeRatesProvider.java]: This file is obsolete in the current version.

File: wallet/src/de/schildbach/wallet/ui/send/FeeCategory.java [hunk 7]
```diff
@@ -27,18 +27,18 @@
 	/**
 	 * We don't care when it confirms, but it should confirm at some time. Can be days or weeks.
 	 */
-	ECONOMIC(Coin.valueOf(1000)), // 0.02 mBTC
+	ECONOMIC(Coin.valueOf(10000)), // 0.02 mBTC
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/send/FeeCategory.java]: This is something we forgot to change, but now that dash fees are at 1000, changes to this file are unnecessary

File: wallet/testNet3/res/values/values.xml [hunk 8]
```diff
@@ -1,7 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (HashEngineering) [inline wallet/testNet3/res/values/values.xml]: Why is this changed?

**Comment by HashEngineering** [comment]
Some notes were made on some of the changes.  Many files have white space changes, which we don’t want to make unless Bitcoin Wallet makes them.

The main thing to be done is to rebase against the current master branch.  There will be a lot of merge conflicts from doing this, but it is necessary.

**Comment by rgenito** [comment]
@HashEngineering thank you for the input. I'll be sure that our devs pay attention to this.

**Comment by HashEngineering** [comment]
closed, since this request has been replaced by others at:
https://github.com/genitrust/dash-wallet/pulls
